test/serializers/associations_test.rb

```diff
@@ -290,6 +290,48 @@ def test_illegal_conditional_associations
           assert_match(/:if should be a Symbol, String or Proc/, exception.message)
         end
       end
+
+      class InheritedSerializerTest < ActiveSupport::TestCase
+        InheritedPostSerializer = Class.new(PostSerializer) do
+          belongs_to :author, polymorphic: true
+          has_many :comments, key: :reviews
+        end
+
+        InheritedAuthorSerializer = Class.new(AuthorSerializer) do
+          has_many :posts, polymorphic: true
+          has_one :roles, key: :new_roles
+          has_one :bio, polymorphic: true
+        end
+
+        def setup
+          @author = Author.new(name: 'Steve K.')
+          @author.bio = Bio.new(id: 1, content: 'AMS Contributor')
+          @blog = Blog.new(name: 'AMS Blog')
+          @post = Post.new(title: 'New Post', body: 'Body')
+          @post.blog = @blog
+          @post.author = @author
+          @post_serializer = PostSerializer.new(@post, custom_options: true)
+          @author_serializer = AuthorSerializer.new(@author)
+          @inherited_post_serializer = InheritedPostSerializer.new(@post, custom_options: true)
+          @inherited_author_serializer = InheritedAuthorSerializer.new(@author)
+        end
+
+        def test_redefined_has_many_and_has_one
+          author_associations = @author_serializer.associations.to_a
+          inherited_author_associations = @inherited_author_serializer.associations.to_a
+          assert_equal(@author_serializer.associations.count, 3)
+          assert_equal(@inherited_author_serializer.associations.count, 4)
+          assert_equal((author_associations - inherited_author_associations).count, 2)
```

Comment:
can you add assertions to this that show what associations are being produced? 
counting associations doesn't actually prove that this is working, just that there are different associations.